packaging/deb/control/postinst

```diff
@@ -14,7 +14,15 @@ case "$1" in
             groupadd -r "$GRAFANA_AGENT_GROUP"
         fi
         if ! getent passwd "$GRAFANA_AGENT_USER" > /dev/null 2>&1 ; then
-            useradd -m -r -g grafana-agent -d /var/lib/grafana-agent -s /sbin/nologin -c "grafana-agent user" grafana-agent
+            useradd -m -r -g "$GRAFANA_AGENT_GROUP" -d /var/lib/grafana-agent -s /sbin/nologin -c "grafana-agent user" "$GRAFANA_AGENT_USER"
+        fi
+
+        # Add grafana agent user to groups used for reading logs.
+        if ! getent group adm > /dev/null 2>&1 ; then
```

Comment:
I think you want to invert the logic here, we want to add the supplementary group if it exists and do nothing otherwise.
Same for all others.

Reply:
ah nice, thanks, good catch